File: backend/schemas/checkpoints.py
"""Agent Checkpoint API 数据结构。"""

from __future__ import annotations

from typing import Any, Literal

from pydantic import Field

from backend.schemas.common import FlexibleModel


class CheckpointCreateBody(FlexibleModel):
    """创建通用 Agent Checkpoint。"""

    session_id: str = Field(alias="sessionId")
    agent_kind: Literal["qa", "code", "media", "commerce"] = Field(
        alias="agentKind"
    )
    route: str
    request: dict[str, Any] = Field(default_factory=dict)
    label: str = ""
    checkpoint_id: str = Field(default="", alias="checkpointId")


class CheckpointUpdateBody(FlexibleModel):
    """更新 Checkpoint 状态或执行快照。"""

    status: Literal[
        "running", "paused", "interrupted", "failed", "completed", "discarded"
    ] | None = None
    state: dict[str, Any] | None = None
    request: dict[str, Any] | None = None
    error_message: str | None = Field(default=None, alias="errorMessage")
    resumable: bool | None = None

File: app/types/checkpoints.ts
import type {
  AttachedFile,
  CodeAgentExecutionMode,
  ComposerMode,
  ImageEditFidelity,
  TypographyPolicy,
} from "../constants/page-constants";
import type { CommerceWorkflowMode } from "../lib/commerce/listing/types";
import type { CommerceMarketplaceCode } from "../lib/commerce/types";

export type AgentCheckpointKind = "qa" | "code" | "media" | "commerce" | "image";
export type AgentCheckpointStatus =
  | "running"
  | "paused"
  | "interrupted"
  | "failed"
  | "completed"
  | "discarded";

export interface AgentCheckpointRequest {
  input: string;
  selectedModel: string;
  composerMode: ComposerMode;
  codeAgentMode: CodeAgentExecutionMode;
  attachments: AttachedFile[];
  commerceWorkflowMode: CommerceWorkflowMode;
  commerceMarketplace: CommerceMarketplaceCode;
  typographyPolicy: TypographyPolicy;
  imageEditFidelity: ImageEditFidelity;
  enableQualityGuard: boolean;
}

export interface AgentCheckpoint {
  id: string;
  sessionId: string;
  agentKind: AgentCheckpointKind;
  route: string;
  status: AgentCheckpointStatus;
  resumable: boolean;
  request: AgentCheckpointRequest;
  state: Record<string, unknown>;
  label: string;
  errorMessage: string;
  createdAt: string;
  updatedAt: string;
  completedAt?: string | null;
}

export interface CheckpointFinishResult {
  status: "completed" | "paused" | "interrupted" | "failed";
  error?: string;
}
